src/NGen.cpp

```diff
@@ -557,13 +573,31 @@ int main(int argc, char *argv[]) {
           
           router->route(number_of_timesteps, delta_time); 
         }
- #ifdef NGEN_MPI_ACTIVE
     }
- #endif //NGEN_MPI_ACTIVE
- #else
- #ifdef NGEN_MPI_ACTIVE
+#endif
+
+    auto time_done_routing = std::chrono::steady_clock::now();
+    auto time_elapsed_routing = time_done_routing - time_done_simulation;
+
+    auto as_seconds = [](decltype(time_elapsed_routing) in)
```

Comment:
move this lambda inside the `if(mpi_rank == 0)`?  Only rank 0 is using it.

Reply:
Sure. I expect it'll eventually end up as a standalone function somewhere in a utility header, but I can keep it super-local for now until it has more usage.